tests/models/test_dagrun.py

```diff
@@ -552,3 +561,131 @@ def with_all_tasks_removed(dag):
         dagrun.verify_integrity()
         flaky_ti.refresh_from_db()
         self.assertEqual(State.NONE, flaky_ti.state)
+
+    def test_depends_on_past(self):
+        # dag_id = 'test_depends_on_past'
```

Comment:
After getting clarification on this, I will delete the comments in the tests

Reply:
If the test don't take a significant amount of time to run then we can have these tests

Reply:
Oh, actually, the tests that I included are the modified versions of the original tests. The main difference are as follow:
1. Original Author (OA) uses `dag.run()`, while I uses `task_instance.run()`

Reply:
Scheduler deadlock in which case, there are known issues of Scheduler deadlock if MySql. Other than that I am not sure if there are any

Reply:
yes, remove the comments in the PR, should be fine

Reply:
OK got it. Since the tests pass anyways without any scheduler deadlock, there shouldn't be any problem. I will remove the comments